source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeLinksCommands/Validation/BusinessValidation/ValidationRules/ChargeLinksUpdateNotYetSupportedRule.cs

```diff
@@ -41,16 +42,16 @@ public class ChargeLinksUpdateNotYetSupportedRule : IValidationRuleWithExtendedD
 
         private bool ChargeLinkDateRangeIsNotOverlapping(ChargeLinkDto newLink)
         {
-            foreach (var existingLink in _existingChargeLinks)
-            {
-                // See https://stackoverflow.com/questions/325933/determine-whether-two-date-ranges-overlap
-                if (existingLink.StartDateTime < newLink.EndDateTime && existingLink.EndDateTime > newLink.StartDateTime)
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return _existingChargeLinks.All(existingLink => !IsOverlapping(existingLink, newLink));
+        }
+
+        private static bool IsOverlapping(ChargeLink existingLink, ChargeLinkDto newLink)
+        {
+            // See https://stackoverflow.com/questions/325933/determine-whether-two-date-ranges-overlap
+            var isOverlapping = (existingLink.StartDateTime < newLink.EndDateTime || newLink.EndDateTime == null)
```

Comment:
nit, but in terms of evaluation I would feel safer with this (I guess that - at least in theory - Instant < null could return true):
```suggestion
            var isOverlapping = (newLink.EndDateTime == null || existingLink.StartDateTime < newLink.EndDateTime)
```